pkg/overlay/server.go

```diff
@@ -37,59 +33,100 @@ func (o *Server) Lookup(ctx context.Context, req *proto.LookupRequest) (*proto.L
 	}
 
 	return &proto.LookupResponse{
-		Node: &proto.Node{
-			Id:      req.GetNodeID(),
-			Address: na,
-		},
+		Node: na,
 	}, nil
 }
 
 // FindStorageNodes searches the overlay network for nodes that meet the provided requirements
-func (o *Server) FindStorageNodes(ctx context.Context, req *proto.FindStorageNodesRequest) (*proto.FindStorageNodesResponse, error) {
-	// NB:  call FilterNodeReputation from node_reputation package to find nodes for storage
-	keys, err := o.cache.DB.List(nil, storage.Limit(10))
+func (o *Server) FindStorageNodes(ctx context.Context, req *proto.FindStorageNodesRequest) (resp *proto.FindStorageNodesResponse, err error) {
```

Comment:
This method is called by the Overlay Client. Is it in scope of this PR to make it pass the `Limit` option?

Also the name `Limit` implies that the result can contain less than the `Limit` number of nodes. The Segment Store calls `Overlay.Choose` to get a list of nodes to store the piece in. It's crucial that it gets exactly the number of requested nodes.

Reply:
Yes, That's a good call, I'll add that.

It will return less than the specified number of nodes in limit only if it doesn't have enough nodes to provide. There is no way around this as far as I can see. If I only have 5 nodes that can store what you are asking for and you want 10, I can't make up an extra 5.

Reply:
Then better return an error?

Or do we have a use case where a caller can work with less than the requested number of nodes? The ECClient cannot for sure.

Reply:
I would say it is up to the caller to decide what they want to do if they get less than the requested limit